assets/js/frontend/add-to-cart-variation.js

```diff
@@ -1,238 +1,218 @@
+			form.variationData = form.$form.data( 'product_variations' );
+			form.useAjax       = false === form.variationData;
+			form.$form.trigger( 'check_variations' );
+		},
 
-		// When the variation is hidden
-		.on( 'hide_variation', function( event ) {
+		/**
+		 * When a variation is hidden.
+		 */
+		onHide: function( event ) {
 			event.preventDefault();
-			$form.find( '.single_add_to_cart_button' ).removeClass( 'wc-variation-is-unavailable' ).addClass( 'disabled wc-variation-selection-needed' );
-			$form.find( '.woocommerce-variation-add-to-cart' ).removeClass( 'woocommerce-variation-add-to-cart-enabled' ).addClass( 'woocommerce-variation-add-to-cart-disabled' );
-		} )
+			event.data.variationForm.$form.find( '.single_add_to_cart_button' ).removeClass( 'wc-variation-is-unavailable' ).addClass( 'disabled wc-variation-selection-needed' );
+			event.data.variationForm.$form.find( '.woocommerce-variation-add-to-cart' ).removeClass( 'woocommerce-variation-add-to-cart-enabled' ).addClass( 'woocommerce-variation-add-to-cart-disabled' );
+		},
 
-		// When the variation is revealed
-		.on( 'show_variation', function( event, variation, purchasable ) {
+		/**
+		 * When a variation is shown.
+		 */
+		onShow: function( event, variation, purchasable ) {
 			event.preventDefault();
 			if ( purchasable ) {
-				$form.find( '.single_add_to_cart_button' ).removeClass( 'disabled wc-variation-selection-needed wc-variation-is-unavailable' );
-				$form.find( '.woocommerce-variation-add-to-cart' ).removeClass( 'woocommerce-variation-add-to-cart-disabled' ).addClass( 'woocommerce-variation-add-to-cart-enabled' );
+				event.data.variationForm.$form.find( '.single_add_to_cart_button' ).removeClass( 'disabled wc-variation-selection-needed wc-variation-is-unavailable' );
+				event.data.variationForm.$form.find( '.woocommerce-variation-add-to-cart' ).removeClass( 'woocommerce-variation-add-to-cart-disabled' ).addClass( 'woocommerce-variation-add-to-cart-enabled' );
 			} else {
-				$form.find( '.single_add_to_cart_button' ).removeClass( 'wc-variation-selection-needed' ).addClass( 'disabled wc-variation-is-unavailable' );
-				$form.find( '.woocommerce-variation-add-to-cart' ).removeClass( 'woocommerce-variation-add-to-cart-enabled' ).addClass( 'woocommerce-variation-add-to-cart-disabled' );
+				event.data.variationForm.$form.find( '.single_add_to_cart_button' ).removeClass( 'wc-variation-selection-needed' ).addClass( 'disabled wc-variation-is-unavailable' );
+				event.data.variationForm.$form.find( '.woocommerce-variation-add-to-cart' ).removeClass( 'woocommerce-variation-add-to-cart-enabled' ).addClass( 'woocommerce-variation-add-to-cart-disabled' );
 			}
-		} )
-
-		.on( 'click', '.single_add_to_cart_button', function( event ) {
-			var $this = $( this );
+		},
 
-			if ( $this.is('.disabled') ) {
+		/**
+		 * When the cart button is pressed.
+		 */
+		onAddToCart: function( event ) {
+			if ( $( this ).is('.disabled') ) {
```

Comment:
Nitpicking again. Search and replace `('` and `')` with `( '` and `' )` for consistency?